src/appdistribution/client.ts

```diff
@@ -0,0 +1,154 @@
+import * as _ from "lodash";
+import * as api from "../api";
+import * as utils from "../utils";
+import { Distribution } from "./distribution";
+import { FirebaseError } from "../error";
+
+// tslint:disable-next-line:no-var-requires
+const pkg = require("../../package.json");
+
+/**
+ * Proxies HTTPS requests to the App Distribution server backend.
+ */
+export class AppDistributionClient {
+  static MAX_POLLING_RETRIES = 30;
+  static POLLING_INTERVAL_MS = 1000;
+
+  constructor(private readonly appId: string) {
+    this.appId = appId;
```

Comment:
Using `constructor(private readonly appId: string)` eliminates the need for `this.appId = appId`

https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties